policy-engine/src/graph.rs

```diff
@@ -40,59 +39,65 @@ lazy_static! {
 }
 
 /// Serve Cincinnati graph requests.
-pub(crate) fn index(req: HttpRequest<AppState>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+pub(crate) fn index(
+    req: HttpRequest<AppState>,
+) -> Box<Future<Item = HttpResponse, Error = GraphError>> {
     HTTP_GRAPH_REQS.inc();
 
+    // Check that the client can accept JSON media type.
+    if let Err(e) = commons::ensure_content_type(req.headers(), CONTENT_TYPE) {
```

Comment:
opportunity: combine this into something declarative with the next error check

Reply:
I'm not sure I got what you are suggesting here. Would you like a chain of `and_then` checks or something different? I won't mind if you could very very briefly sketch what you have in mind with "something declarative".

Reply:
I thought about something like:

```rust
    for checks in vec![
        // Check that the client can accept JSON media type.
        || commons::ensure_content_type(req.headers(), CONTENT_TYPE),
        // Check for required client parameters.
        || commons::ensure_query_params(mandatory_params, req.query_string()),
    ] {
        if let Err(e) = check() {
            HTTP_GRAPH_BAD_REQS.inc();
            return Box::new(future::err(e));
        }
    }
```

I tried this quickly but it's broken due to closures not matching in type.